pkg/apikey/apikey.go

```diff
@@ -61,123 +57,76 @@ func New(organizationID string) (*Key, error) {
-		// Even Y
-		prefix = "02"
-	} else {
-		// Odd Y
-		prefix = "03"
+	// generate key pair data
+	switch apiKey.scheme {
+	case SchemeP256:
+		apiKey, err = newECDSAKey(apiKey.scheme)
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate p256 key pair: %s", err)
+		}
+	case SchemeSECP256K1:
+		apiKey, err = newECDSAKey(apiKey.scheme)
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate secp256k1 key pair: %s", err)
+		}
+	case SchemeED25519:
+		apiKey, err = newED25519Key()
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate ed25519 key pair: %s", err)
+		}
+	default:
+		return nil, fmt.Errorf("unsupported signature scheme: %s", apiKey.scheme)
 	}
 
-	// Encode the public key X coordinate as 64 hexadecimal characters, padded with zeroes as necessary
-	return fmt.Sprintf("%s%064x", prefix, publicKey.X)
-}
-
-// FromECDSAPrivateKey takes an ECDSA keypair and forms a Turnkey API key from it.
-// Assumes that privateKey.PublicKey has already been derived.
-func FromECDSAPrivateKey(privateKey *ecdsa.PrivateKey) (*Key, error) {
-	if privateKey == nil || privateKey.PublicKey.X == nil {
-		return nil, errors.New("empty key")
-	}
-
-	publicKey := &privateKey.PublicKey
-
-	return &Key{
-		TkPrivateKey: EncodePrivateKey(privateKey),
-		TkPublicKey:  EncodePublicKey(publicKey),
-		publicKey:    publicKey,
-		privateKey:   privateKey,
-	}, nil
-}
-
-// FromTurnkeyPrivateKey takes a Turnkey-encoded private key, derives a public key from it, and then returns the corresponding Turnkey API key.
-func FromTurnkeyPrivateKey(encodedPrivateKey string) (*Key, error) {
-	bytes, err := hex.DecodeString(encodedPrivateKey)
-	if err != nil {
-		return nil, err
-	}
-
-	dValue := new(big.Int).SetBytes(bytes)
-
-	publicKey := new(ecdsa.PublicKey)
-	privateKey := ecdsa.PrivateKey{
-		PublicKey: *publicKey,
-		D:         dValue,
-	}
-
-	// Derive the public key
-	privateKey.PublicKey.Curve = elliptic.P256()
-	privateKey.PublicKey.X, privateKey.PublicKey.Y = privateKey.PublicKey.Curve.ScalarBaseMult(privateKey.D.Bytes())
-
-	apiKey, err := FromECDSAPrivateKey(&privateKey)
-	if err != nil {
-		return nil, err
-	}
+	// supply metadata
+	apiKey.Metadata.Organizations = append(apiKey.Metadata.Organizations, organizationID)
+	apiKey.Metadata.PublicKey = apiKey.PublicKey
+	apiKey.Metadata.Scheme = string(apiKey.scheme)
 
 	return apiKey, nil
 }
 
-// DecodeTurnkeyPublicKey takes a Turnkey-encoded public key and creates an ECDSA public key.
-func DecodeTurnkeyPublicKey(encodedPublicKey string) (*ecdsa.PublicKey, error) {
-	bytes, err := hex.DecodeString(encodedPublicKey)
-	if err != nil {
-		return nil, err
+// FromTurnkeyPrivateKey takes a Turnkey-encoded private key, derives a public key from it, and then returns the corresponding Turnkey API key.
+func FromTurnkeyPrivateKey(encodedPrivateKey string, scheme signatureScheme) (*Key, error) {
+	if scheme == SchemeED25519 {
+		return fromTurnkeyED25519Key(encodedPrivateKey)
```

Comment:
Is this maybe leftover from some previous version?  This case is handled in the following `switch` clause, no?